### PR TITLE
Replace in batch.

### PR DESCRIPTION
**Pull request by geekbeast**

### CODE REVIEW

File: src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt [hunk 1]
```diff
@@ -203,72 +203,79 @@ class OrganizationMetadataEntitySetsService(private val edmService: EdmManager)
...
+                    propertyTypes.getValue(ID).id to setOf(propertyType.id.toString()),
+                    propertyTypes.getValue(DATASET_NAME).id to setOf(entitySet.name),
+                    propertyTypes.getValue(COL_NAME).id to setOf(propertyType.type.fullQualifiedNameAsString),
+                    propertyTypes.getValue(ORG_ID).id to setOf(organizationId.toString()),
+                    propertyTypes.getValue(TYPE).id to setOf(propertyType.datatype.toString()),
+                    propertyTypes.getValue(DESCRIPTION).id to setOf(propertyType.description)
+            )
 
+            val columnEntityKeyId = getColumnEntityKeyId(organizationMetadataEntitySetIds, entitySet.id, propertyType)
+            dataGraphManager.partialReplaceEntities(
```
Comment (katherinebernstein) [inline src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt:225]: I think we should probably pull this out of the forEach, so like have the forEach construct the map of ekids to entities, and then do one batch upsert using `partialReplaceEntities`

File: src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt [hunk 2]
```diff
@@ -203,72 +203,79 @@ class OrganizationMetadataEntitySetsService(private val edmService: EdmManager)
...
-        val columnEntity = mutableMapOf<UUID, Set<Any>>(
-                propertyTypes.getValue(ID).id to setOf(column.id.toString()),
-                propertyTypes.getValue(DATASET_NAME).id to setOf(table.name),
-                propertyTypes.getValue(COL_NAME).id to setOf(column.name),
-                propertyTypes.getValue(ORG_ID).id to setOf(organizationId.toString()),
-                propertyTypes.getValue(TYPE).id to setOf(column.dataType.toString()),
-                propertyTypes.getValue(DESCRIPTION).id to setOf(column.description)
-        )
+        val columnEntities = columns.map { column ->
+            val columnEntity = mutableMapOf<UUID, Set<Any>>(
```
Comment (katherinebernstein) [inline src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt:252]: same as above

File: src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt [hunk 3]
```diff
@@ -203,72 +203,79 @@ class OrganizationMetadataEntitySetsService(private val edmService: EdmManager)
         )
     }
 
-    fun addDatasetColumn(entitySet: EntitySet, propertyType: PropertyType) {
+    fun addDatasetColumn(entitySet: EntitySet, propertyTypesInEntitySet: Collection<PropertyType>) {
```
Comment (katherinebernstein) [inline src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt:206]: would also be good to rename these both to `addDatasetColumn` while we're making updates here in all the projects anyway

File: src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt (hunk 1 above)
Comment (geekbeast) [reply src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt]: good idea, much cleaner too.

File: src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt (hunk 3 above)
Comment (geekbeast) [reply src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt:206]: :+1:

Comment (geekbeast) [reply src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt:206]: I assumed you meant addDatasetColumns here.

File: src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt [hunk 4]
```diff
@@ -203,72 +203,72 @@ class OrganizationMetadataEntitySetsService(private val edmService: EdmManager)
...
-                mapOf(columnEntityKeyId to columnEntity),
+                propertyTypeEntities,
                 columnAuthorizedPropertTypes
         )
 
+        val datasetEntityKeyId = getDatasetEntityKeyId(organizationMetadataEntitySetIds, entitySet.id)
+        val datasetColumnEntities = mutableMapOf<UUID, Set<Any>>(
+                propertyTypes.getValue(COL_INFO).id to setOf(mapper.writeValueAsString(propertyTypeEntities.values))
+        )
         dataGraphManager.mergeEntities(
```
Comment (katherinebernstein) [inline src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt:235]: this should probably be a PartialReplace now too

Comment (geekbeast) [reply src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt:235]: Weird, I thought you had changed that one. Fixed.

**Review by katherinebernstein — APPROVED** [review]

File: src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt (hunk 4 above)
Comment (katherinebernstein) [reply src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt]: I thought I had too ¯\_(ツ)_/¯

**Pull request merged**